### PR TITLE
Fix Infinite loop in BLE disconnect

### PR DESCRIPTION
**Pull request by leegeth**
Fix Infinite loop in BLE disconnect

Description
-----------
Shadow demo and MQTT demo on Nordic were not completing on Nordic.
The root cause was that the demo was completing but it was ending up in
an infinite loop in BLE disconnect logic and thus blocking the lower
priority logging task . This prevented logs of demo completion to be not
printed.
Solution is done by exiting the infinite loop in a condition when there
is error.

Checklist:
----------
- [ x] I have tested my changes.

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: libraries/c_sdk/standard/ble/src/iot_ble_gap.c [hunk 1]
```diff
@@ -514,9 +514,15 @@ BTStatus_t IotBle_Off( void )
...
             if( pConnInfo != NULL )
             {
-                ( void ) _BTInterface.pBTLeAdapterInterface->pxDisconnect( _BTInterface.adapterIf,
+                status = _BTInterface.pBTLeAdapterInterface->pxDisconnect( _BTInterface.adapterIf,
                                                                            &bdAddr,
                                                                            connId );
+
+                if( status != eBTStatusSuccess )
+                {
+                    IotLogError( "Failed disconnect" );
```
Comment (qiutongs) [inline libraries/c_sdk/standard/ble/src/iot_ble_gap.c]: Nitpick, can you also log the status?

Comment (leegeth) [reply libraries/c_sdk/standard/ble/src/iot_ble_gap.c]: Done

**Comment by dan4thewin** [comment]
To benefit future readers, it might be nice to have a comment about the life cycle of pConnInfo, i.e., does disconnect remove it from the linked list, etc.

**Comment by leegeth** [comment]
> To benefit future readers, it might be nice to have a comment about the life cycle of pConnInfo, i.e., does disconnect remove it from the linked list, etc.

Sure. That's a good suggestion. I have talked to Ravi about it and we will create a SIM to handle it.

**Pull request merged**